clients/unshaded/src/main/java/tachyon/client/netty/NettyRemoteBlockReader.java

```diff
@@ -38,11 +39,13 @@
 /**
  * Read data from remote data server using Netty.
  */
-public final class NettyRemoteBlockReader implements RemoteBlockReader {
+public final class NettyRemoteBlockReader implements Closeable, RemoteBlockReader {
```

Comment:
If RemoteBlockReader implements Closeable, then you don't have to check for:

instanceof NettyRemoteBlockReader

when you close the reader.